examples/async/index.js

```diff
@@ -1,15 +1,43 @@
 import 'babel-polyfill'
 import React from 'react'
-import { render } from 'react-dom'
+import ReactDOM from 'react-dom'
 import { Provider } from 'react-redux'
-import App from './containers/App'
 import configureStore from './store/configureStore'
 
 const store = configureStore()
+const rootEl = document.getElementById('root')
 
-render(
-  <Provider store={store}>
-    <App />
-  </Provider>,
-  document.getElementById('root')
-)
+let render = () => {
+  const App = require('./containers/App').default
```

Comment:
I've been following the discussion about vanilla HMR and decided to use that approach when upgrading to webpack in my [universal-dev-toolkit](https://github.com/stoikerty/universal-dev-toolkit).

I'm using `react-router` with a separate `routes`-file and a `RouterContainer` that serves as the root-component which is reloaded [in the `app.js`-file](https://github.com/stoikerty/universal-dev-toolkit/blob/master/src/client/app.js) using the approach in this PR. It all seems to work well for me so far. When I add a new view to the router, everything refreshes accordingly. Perhaps this could serve as a starting point for someone wishing to implement it themselves?

If you want to add Redux into the toolkit, you can wrap the `<Provider store={store}>` around the `RouterContainer` like @0x80 did (see previous comment).